core/EditingTools/gridAndLabelCreator.py

```diff
@@ -1,425 +1,716 @@
+			y = y1.y()
+			testif = abs(floor(abs(round(y1.y(), 4) - (y_min % (py)) - (0.002 * map_scale))/py) - floor(abs(round(y1.y(), 4) - (y_min % (py)))/py))
+			if testif >= 1:
+				y = y + dy1
+			else:
+				testif2 = abs(floor(abs(round(y1.y(), 4) - (y_min % (py)))/py) - floor(abs(round(y1.y(), 4) - (y_min % (py)) + (0.002 * map_scale))/py))
+				if testif2 >= 1:
+					y = y + dyO
+				else:
+					y = y + dy
+
+			full_label = str((floor(y_UTM/grid_spacing)+u)*grid_spacing)
+			if label_index == 1:
+				expression_str = full_label[ : len( full_label )-5]
+			elif label_index == 2:
+				expression_str = str('\'')+full_label[len( full_label )-5 : -3]+str('\'')
+				fSize = 7.08*fSize/4.25
+			elif label_index == 3:
+				expression_str = str('\'')+full_label[-3 : ]+str('\'')
+
+		# Label Format Settings
+		settings = QgsPalLayerSettings()
+		settings.Placement = QgsPalLayerSettings.Free
+		settings.isExpression = True
+		textprop = QgsTextFormat()
+		textprop.setColor(QColor(0,0,0,255))
+		textprop.setSize(fSize)
+		textprop.setFont(QFont(fontType))
+		textprop.setLineHeight(1)
+		settings.setFormat(textprop)
+
+		# Label Name and Position
+		settings.fieldName = expression_str
+		datadefined = QgsPropertyCollection()
+		datadefined.setProperty(9, x)
+		datadefined.setProperty(10, y)
+		if not(hAlign == ''):
+			datadefined.setProperty(11, hAlign)
+		if not(vAlign == ''):
+			datadefined.setProperty(12, vAlign)
+		# Creating and Activating Labeling Rule
+		settings.setDataDefinedProperties(datadefined)
+		rule = QgsRuleBasedLabeling.Rule(settings)
+		rule.setDescription(desc)
+		rule.setActive(True)
+		#Exit: Label Rule
+		return rule
+
+
+	def Conv_dec_gms (base_coord, coord_spacing, u, neg_character, pos_character):
+		
+		x = base_coord + coord_spacing*u
+		conv_exp_str = 'concat(floor(round(abs('+str(x)+'),4)),'+str('\'º\'')+','+str('\' \'')+', if(round((round((-floor(round(abs('+str(x)+'),4))+round(abs('+str(x)+'),4)),4)*60-floor(round((-floor(round(abs('+str(x)+'),4))+round(abs('+str(x) \
+			+'),4)),4)*60))*60) = 60, concat(floor(round((-floor(round(abs('+str(x)+'),4))+round(abs('+str(x)+'),4)),4)*60)+1,'+str('\' 0\'\'\'')+'), concat(floor(round((-floor(round(abs('+str(x)+'),4))+round(abs('+str(x)+'),4)),4)*60),'+str('\'"\'') \
+			+','+str('\' \'')+',round((round((-floor(round(abs('+str(x)+'),4))+round(abs('+str(x)+'),4)),4)*60-floor(round((-floor(round(abs('+str(x)+'),4))+round(abs('+str(x)+'),4)),4)*60))*60),'+str('\'\'\'\'')+')),if('+str(x) \
+			+'<0, '+str('\' ')+neg_character+str('\'')+','+str('\' ')+pos_character+str('\'')+'))'
+		
+		return conv_exp_str
+
+
+
+	def styleCreator (layer, index, spacing, crossX, crossY, scale, color, fontSize, font):
+		
+		grid_spacing = spacing
+		geo_number_x = crossX
+		geo_number_y = crossY
+		map_scale = scale*1000
+		grid_color = color
+		fSize = fontSize
+		fontType = font
+		
+		#Loading feature
+		layer_bound = layer
+		feature_bound = layer_bound.getFeature(0)

```

Comment:
isso dá problemas, não é a forma mais adequada de pegar feição. O melhor é fazer como estava antes:

layer_bound = [i for i in layer.getFeatures()][0]

Além disso, como seria o caso de uso ter várias molduras para criar grid? Pergunto isso, pois será o caso de impressão de produtos em lote.